src/storage/storage_util.cpp

```diff
@@ -6,51 +6,15 @@
 #include "storage/projected_columns.h"
 #include "storage/tuple_access_strategy.h"
 #include "storage/undo_record.h"
-
 namespace terrier::storage {
-void StorageUtil::WriteBytes(const uint8_t attr_size, const uint64_t val, byte *const pos) {
-  switch (attr_size) {
-    case sizeof(uint8_t):
-      *reinterpret_cast<uint8_t *>(pos) = static_cast<uint8_t>(val);
-      break;
-    case sizeof(uint16_t):
-      *reinterpret_cast<uint16_t *>(pos) = static_cast<uint16_t>(val);
-      break;
-    case sizeof(uint32_t):
-      *reinterpret_cast<uint32_t *>(pos) = static_cast<uint32_t>(val);
-      break;
-    case sizeof(uint64_t):
-      *reinterpret_cast<uint64_t *>(pos) = static_cast<uint64_t>(val);
-      break;
-    default:
-      // Invalid attr size
-      throw std::runtime_error("Invalid byte write value");
-  }
-}
-
-uint64_t StorageUtil::ReadBytes(const uint8_t attr_size, const byte *const pos) {
-  switch (attr_size) {
-    case sizeof(uint8_t):
-      return *reinterpret_cast<const uint8_t *>(pos);
-    case sizeof(uint16_t):
-      return *reinterpret_cast<const uint16_t *>(pos);
-    case sizeof(uint32_t):
-      return *reinterpret_cast<const uint32_t *>(pos);
-    case sizeof(uint64_t):
-      return *reinterpret_cast<const uint64_t *>(pos);
-    default:
-      // Invalid attr size
-      throw std::runtime_error("Invalid byte write value");
-  }
-}
 
 template <class RowType>
 void StorageUtil::CopyWithNullCheck(const byte *const from, RowType *const to, const uint8_t size,
                                     const uint16_t projection_list_index) {
   if (from == nullptr)
     to->SetNull(projection_list_index);
   else
-    WriteBytes(size, ReadBytes(size, from), to->AccessForceNotNull(projection_list_index));
+    TERRIER_MEMCPY(to->AccessForceNotNull(projection_list_index), from, size);
```

Comment:
I'll probably revert this back to plain `memcpy` since I am convinced that the results just came back that way because of cpu scaling. Thanks for catching it though.